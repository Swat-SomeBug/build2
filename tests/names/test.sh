#!/bin/sh

valgrind -q b -q | diff -u test.out -
