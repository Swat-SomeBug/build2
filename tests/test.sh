#! /usr/bin/env bash

cur_dir="`pwd`"
trap 'cd "$cur_dir"' EXIT

export PATH=$cur_dir/../build2:$PATH

function test ()
{
  echo "Testing $1"
  cd "$cur_dir/$1"
  ./test.sh
}

test "amalgam/unnamed"
test "escaping"
test "eval"
test "function/call"
test "if-else"
test "keyword"
test "names"
test "pairs"
test "quote"
test "scope"
test "variable/expansion"
test "variable/null"
test "variable/override"
test "variable/prepend"
test "variable/qualified"
test "variable/type"
