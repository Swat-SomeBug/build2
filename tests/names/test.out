foo bar baz
fox/ bar baz
fox/foo bar baz
bar
foo{}
foo{}
foo{bar}
foo{bar} foo{baz}
foo{bar} fox{bar}
foo{bar} fox{bar} foo{baz} fox{baz}
dir/bar
foo{dir/}
foo{dir/}
dir/foo{bar}
dir/foo{bar} dir/foo{baz}
dir/foo{bar} dir/fox{bar}
dir/foo{bar} dir/fox{bar} dir/foo{baz} dir/fox{baz}
dir/bar
dir/bar dir/baz
dir/bar dor/bar
dir/bar dor/bar dir/baz dor/baz
dir/foo{bar}
dir/foo{bar} dir/foo{baz}
dir/foo{bar} dor/fox{bar}
dir/foo{bar} dor/fox{bar} dir/foo{baz} dor/fox{baz}
dir/foo{bar}
dir/foo{bar} dir/foo{baz}
dir/foo{bar} dor/foo{bar}
dir/foo{bar} dor/foo{bar} dir/fox{bar} dor/fox{bar} dir/foo{baz} dor/foo{baz} dir/fox{baz} dor/fox{baz}
prj%foo{bar} prj%foo{baz}
foo{bar} prj%foo{baz}
dir/foo{bar} dir/foo{baz}
foo{bar} dir/foo{baz}
dir/foo{bar} dir/dor/foo{baz}
dir/foo{bar}
dir/foo{bar}
dir/foo{bar}
